### PR TITLE
Familiarize entity names before sending them to GitHub

### PR DESCRIPTION
**Pull request by fearphage**
## Problem

I'm taking input from the `EntityPicker` to select groups and users that are then passed to the `publish:github` step as `collaborators`. The problem I'm having is `EntityPicker` returns the values in the form `group:default/group-name` and `user:user-name`. 

## Solution in this PR

These types and namespaces are *exclusively* used by Backstage. They have no meaning to GitHub, GitLab, etc. so there is no value in sending those invalid API requests to those endpoints (since usernames and groups can't contain `:` and `/` for instance). This attempts to resolve that.  

I put this in helpers instead of GitHub utils, because I suspect this will be useful for other vendors besides GitHub.

## Background

I know there's a related ticket #17091 that allows users to manually remove these in some cases (not sure how this applies to arrays of entities). My thinking for this change is that I did not add nor request that Backstage add the type and namespace to the entity so it feels a bit counterintuitive that I have to tell it to remove something that it implicitly added.

### Note

While committing this I received ESLint errors from unchanged code:

```shell
✖ eslint --fix:

/home/fearphage/src/git/backstage/plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.test.ts
  18:1  error  @backstage/plugin-scaffolder does not export src/components/fields/EntityNamePicker  @backstage/no-forbidden-package-imports

✖ 1 problem (1 error, 0 warnings)
````

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by planeiii-te** [comment]
FYI `yarn install` has been running for almost 6 hours now even though there are no changes to `package*.json` files.

¯\\\_(ツ)\_/¯

image

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts [hunk 1]
```diff
@@ -296,3 +296,7 @@ export function getGitCommitMessage(
     ? gitCommitMessage
     : config.getOptionalString('scaffolder.defaultCommitMessage');
 }
+
+export function familiarizeEntityName(name: string): string {
```
Comment (benjdlambert) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts:300]: Think we should call this `entityRefToName` or something instead.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Thanks for this! Think this makes sense.

I think that we should also look at being able to pass in a method that can do this too so if people need to look up the names in a different way they can do that using a method, but think we can skip that for now.

Just a comment about the method name and maybe calling it something slightly different.

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts (hunk 1 above)
Comment (fearphage) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts]: Renamed.

**Review by benjdlambert — APPROVED** [review]
Thanks! :pray:

**Pull request merged**